packages/p2p-v2/src/hooks/__tests__/useQueryString.spec.tsx

```diff
@@ -0,0 +1,232 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import { Router, Route, Switch } from 'react-router-dom';
+import useQueryString from '../useQueryString';
+import { createMemoryHistory } from 'history';
+
+const MockBuySell = () => <h1>Buy sell page</h1>;
+const MockMyProfile = () => {
+    const { deleteQueryString, queryString, replaceQueryString, setQueryString } = useQueryString();
+
+    return (
+        <div>
+            <button
+                onClick={() =>
+                    setQueryString({
+                        tab: 'Payment methods',
+                    })
+                }
+            >
+                Switch payment methods
+            </button>
+            <button
+                onClick={() =>
+                    setQueryString({
+                        tab: 'Stats',
+                    })
+                }
+            >
+                Switch stats
+            </button>
+            <button
+                onClick={() =>
+                    setQueryString({
+                        form: 'Payment method form',
+                        tab: 'Payment methods',
+                    })
+                }
+            >
+                Switch payment method form
+            </button>
+            <button
+                onClick={() =>
+                    replaceQueryString({
+                        form: 'Payment method form',
+                        tab: 'Counterparties',
+                    })
+                }
+            >
+                Switch counterparties
+            </button>
+            <button onClick={() => deleteQueryString('tab')}>Go back</button>
+            {!queryString.get('tab') && <h1>My profile</h1>}
+            {queryString.get('tab') === 'Stats' && <h1>Stats tab</h1>}
+            {queryString.get('tab') === 'Counterparties' && <h1>Counterparties tab</h1>}
+            {queryString.get('tab') === 'Payment methods' && <h1>Payment methods tab</h1>}
+            {queryString.get('form') === 'Payment method form' && <h1>Payment method form</h1>}
+        </div>
+    );
+};
+
+const MockHome = () => <h1>Home</h1>;
+
+const MockAppContent = () => {
+    return (
+        <Switch>
+            <Route component={MockHome} exact path='/' />
+            <Route component={MockBuySell} path='/mock-buy-sell' />
+            <Route component={MockMyProfile} path='/mock-my-profile' />
+        </Switch>
+    );
+};
+
+const mockReplace = jest.fn();
+
+jest.mock('react-router-dom', () => ({
+    ...jest.requireActual('react-router-dom'),
+    useHistory: () => ({
+        push: jest.fn(),
+        replace: mockReplace,
+    }),
+}));
+
+let windowLocationSpy: jest.SpyInstance<Location, []>;
+
+describe('useQueryString', () => {
+    beforeEach(() => {
+        windowLocationSpy = jest.spyOn(window, 'location', 'get');
+    });
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+    it('setting new query strings', () => {
```

Comment:
need to write with 'should' at first so something like this 😅 

```suggestion
    it('should set new query strings', () => {
```